### PR TITLE
Proper handling of AuthenticationFailure

### PR DESCRIPTION
**Pull request by pontusmelke**

### CODE REVIEW

File: community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java [hunk 1]
```diff
@@ -84,7 +84,8 @@ public State init( SessionStateMachine ctx, String clientName, Map<String,Object
                         }
                         catch ( AuthenticationException e )
                         {
-                            return error( ctx, new Neo4jError( e.status(), e.getMessage(), e ) );
+                            ctx.error( new Neo4jError( e.status(), e.getMessage() ));
+                            return halt( ctx );
                         }
                         catch ( Throwable e )
                         {
```
Comment (henriknyman) [inline community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java:91]: What about this case, how can we know that authentication was successful?

Comment (pontusmelke) [inline community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java:91]: nice catch!

File: community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java [hunk 2]
```diff
@@ -84,11 +84,13 @@ public State init( SessionStateMachine ctx, String clientName, Map<String,Object
                         }
                         catch ( AuthenticationException e )
                         {
-                            return error( ctx, new Neo4jError( e.status(), e.getMessage(), e ) );
+                            ctx.error( new Neo4jError( e.status(), e.getMessage() ));
+                            return halt( ctx );
```
Comment (cleishm) [inline community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java:88]: I believe making this `return UNINITIALIZED` is sufficient to resolve the concern raised in #7356.

Comment (cleishm) [inline community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java:88]: @pontusmelke correctly pointed out (in #7356) that I'm mistaken and it has to go to the ERROR state first (or an equivalent new state). So scratch this suggestion.

**Comment by henriknyman** [comment]
I think this is a better solution: https://github.com/neo4j/neo4j/pull/7391

**Comment by pontusmelke** [comment]
Let's go with https://github.com/neo4j/neo4j/pull/7391 instead

**Pull request closed without merging**